src/librustc/lint/builtin.rs

```diff
@@ -590,7 +590,17 @@ impl LintPass for RawPointerDerive {
             return
         }
         let did = match item.node {
-            ast::ItemImpl(..) => {
+            ast::ItemImpl(_, _, _, ref t_ref_opt, _, _) => {
+                // Deriving the Copy trait does not cause a warning
+                if let &Some(ref t_ref) = t_ref_opt {
```

Comment:
Currently this code will mean that any trait called `Copy` isn't warned about, while the compiler actually has enough info to do it "perfectly", i.e. only avoid warning for the special `Copy` trait.

```rust
let trait_ref = ty::node_id_to_trait_ref(cx.tcx, t_ref.ref_id);
let def_id = ty::trait_ref_to_def_id(cx.tcx, trait_ref.def_id);
if Some(def_id) == cx.tcx.lang_items.copy_trait() {
    return
}
```

(I'm not 100% certain that will work, and I haven't compiled it.)